app/controllers/provider_interface/reports/diversity_reports_controller.rb

```diff
@@ -0,0 +1,45 @@
+module ProviderInterface
+  module Reports
+    class DiversityReportsController < ProviderInterfaceController
+      attr_reader :diversity_data, :provider
+      before_action :redirect_if_lacking_permission
+
+      def show
+        @provider = current_user.providers.find(provider_id)
+        zip_filename = ProviderInterface::DiversityReportExport.new(provider: provider).call
+
+        respond_to do |format|
+          format.zip do
```

Comment:
This made me wonder whether we really need to zip this report, rather than just return it as CSV. If it's a specific requirement then I guess it's fine but the report is a summary so it's not going to ever be so big that it needs to be zipped and in any case browsers and web servers can zip responses on the wire automatically.

Reply:
This was because each report is different in terms of what it's presenting in the rows and CSV restricts you to a single sheet. So this way there's a separate CSV for each one.